controllers/cheuser/controller.go

```diff
@@ -0,0 +1,344 @@
+package cheuser
+
+import (
+	"context"
+
+	"github.com/devfile/devworkspace-operator/pkg/constants"
+	"github.com/devfile/devworkspace-operator/pkg/infrastructure"
+	org "github.com/eclipse-che/che-operator/api"
+	v1 "github.com/eclipse-che/che-operator/api/v1"
+	"github.com/eclipse-che/che-operator/api/v2alpha1"
+	"github.com/eclipse-che/che-operator/controllers/che"
+	"github.com/eclipse-che/che-operator/controllers/devworkspace"
+	"github.com/eclipse-che/che-operator/controllers/devworkspace/defaults"
+	"github.com/eclipse-che/che-operator/pkg/deploy"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	projectv1 "github.com/openshift/api/project/v1"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	userSettingsComponentLabelValue = "user-settings"
+)
+
+type CheUserNamespaceReconciler struct {
+	client         client.Client
+	scheme         *runtime.Scheme
+	namespaceCache namespaceCache
+}
+
+var _ reconcile.Reconciler = (*CheUserNamespaceReconciler)(nil)
+
+var (
+	configMapDiffOpts cmp.Option = cmp.Options{
+		cmpopts.IgnoreFields(corev1.ConfigMap{}, "TypeMeta", "ObjectMeta"),
+	}
```

Comment:
It is out of scope of this PR but we already have `configMapDiffOpts`. It would be nice to reuse
https://github.com/eclipse-che/che-operator/blob/main/pkg/deploy/configmap.go#L23